### ISSUE TITLE
Checks that compare against the .json property of a TF data provider are incorrectly evaluated

### ISSUE THREAD

**Issue by bmorrissirromb** [bridgecrewio/checkov#4540]
**Describe the issue**
Checkov checks that evaluate AWS policies (such as `CKV_AWS_070`, which evaluates bucket policy) do not evaluate correctly if the policy is passed via a reference to the .json property of an `aws_iam_policy_document`.

**Examples**
This bucket should be flagged by `CKV_AWS_070` as allowing all AWS principals. However, it does not.
```hcl
resource "aws_s3_bucket" "b" {
  bucket = "my-tf-test-bucket"
}

resource "aws_s3_bucket_policy" "test" {
  bucket = aws_s3_bucket.b.id
  policy = data.aws_iam_policy_document.test.json
}

data "aws_iam_policy_document" "test" {
  statement {
    principals {
      type        = "AWS"
      identifiers = ["*"]
    }
    actions = ["s3:GetObject"]
    resources = ["${aws_s3_bucket.b.arn}/*"]
  }
}
```

Why does it not get flagged? Because in `CKV_AWS_070`, the `conf['policy'][0]` (a large part of the check's logic) just returns the string literal `data.aws_iam_policy_document.test.json`. The string later causes an exception to be thrown when the check tries to `json.loads` the string, and the default behavior of the check is to pass.

Output:
```
        Guide: https://docs.bridgecrew.io/docs/bc_aws_s3_24
Check: CKV_AWS_70: "Ensure S3 bucket does not allow an action with any Principal"
        PASSED for resource: aws_s3_bucket_policy.test
```

**Version (please complete the following information):**
 - Checkov Version: **2.3.33**
